internal/slicer/report_test.go

```diff
@@ -200,13 +207,41 @@ var reportTests = []struct {
 	add: []sliceAndEntry{
 		{entry: fsutil.Entry{Path: "/file"}, slice: oneSlice},
 	},
-	err: `cannot add path "/file" outside of root "/base/"`,
+	err: `cannot add path: "/file" outside of root "/base/"`,
 }, {
 	summary: "File name has root prefix but without the directory slash",
 	add: []sliceAndEntry{
 		{entry: fsutil.Entry{Path: "/basefile"}, slice: oneSlice},
 	},
-	err: `cannot add path "/basefile" outside of root "/base/"`,
+	err: `cannot add path: "/basefile" outside of root "/base/"`,
+}, {
+	summary: "Add mutated regular file",
+	add: []sliceAndEntry{
+		{entry: sampleFile, slice: oneSlice},
+		{entry: sampleDir, slice: oneSlice},
+	},
+	mutated: []*fsutil.Entry{&sampleFileMutated},
+	expected: map[string]slicer.ReportEntry{
+		"/exampleDir/": {
+			Path:   "/exampleDir/",
+			Mode:   fs.ModeDir | 0654,
+			Slices: map[*setup.Slice]bool{oneSlice: true},
+			Link:   "",
+		},
+		"/exampleFile": {
+			Path:      "/exampleFile",
+			Mode:      0777,
+			Hash:      "exampleFile_hash",
+			Size:      5688,
+			Slices:    map[*setup.Slice]bool{oneSlice: true},
+			Link:      "",
+			Mutated:   true,
+			FinalHash: "exampleFile_hash_changed",
+		}},
+}, {
+	summary: "Mutated paths must be added before",
```

Comment:
I think the summary here could be improved by saying something like `Mutated paths must refer to previously added entries`.